crates/polars-expr/src/expressions/sortby.rs

```diff
@@ -58,10 +58,14 @@ pub(super) fn update_groups_sort_by(
     sort_by_s: &Series,
     options: &SortOptions,
 ) -> PolarsResult<GroupsProxy> {
-    let groups = groups
-        .par_iter()
-        .map(|indicator| sort_by_groups_single_by(indicator, sort_by_s, options))
-        .collect::<PolarsResult<_>>()?;
+    // Will trigger a gather for every group, so rechunk before.
+    let sort_by_s = sort_by_s.rechunk();
+    let groups = POOL.install(|| {
```

Comment:
We forgot to launch on our thread pool, so for every group our thread pool was contending with rayon's default. This improved the single chunk query 7 times on a debug build for me.